### DISCUSSION TITLE
Multi-threaded Build?

### DISCUSSION THREAD

**Discussion by eecue** [sveltejs/kit#8913] [Q&A]
I have a very large application (my personal photo site and blog, with roughly a million total pages). It takes a couple hours to build on an Apple M1 Max (10 core 3.2Ghz w/ 64GB of RAM).  What would it take to speed this up.  I read in the docs that the app is single threaded, would it be hard to change that? Maybe build out `/[slug]` pages in parallel? 

> npm run build  1581.91s user 757.20s system 29% cpu 2:10:34.61 total

**Comment by wwwdepot** [comment]
Pump up [concurrency](https://kit.svelte.dev/docs/configuration#prerender). This may help prerendering multiple pages at a time.

If you're not really using Svelte in your project, you might have better time with Hugo/Gatsby for your "large" site.

**Reply by eecue** [reply]
I am for sure using svelte extensively, changing to a different framework doesn't sound like a good solution at this point.

**Reply by eecue** [reply]
Setting concurrency to 8 reduced it a smidge:

> npm run build  1686.10s user 755.32s system 37% cpu 1:48:02.08 total

```JavaScript
// import adapter from '@sveltejs/adapter-auto';
import adapter from '@sveltejs/adapter-static';
import { vitePreprocess } from '@sveltejs/kit/vite';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	// Consult https://kit.svelte.dev/docs/integrations#preprocessors
	// for more information about preprocessors
	preprocess: vitePreprocess(),

	kit: {
		adapter: adapter({
			// default options are shown. On some platforms
			// these options are set automatically — see below
			// pages: 'build',
			// assets: 'build',
			fallback: null,
			precompress: false,
			strict: false
		  }),	
		  prerender: {	  
			concurrency: 8
		  }
	}
};

export default config;

**Reply by wwwdepot** [reply]
See https://github.com/sveltejs/kit/issues/2369 & https://github.com/sveltejs/kit/issues/661#issuecomment-1416205936 on ISR support.
